### PR TITLE
Handle JSON better

### PR DESCRIPTION
**Pull request by gwynne**
**These changes are now available in [4.4.2](https://github.com/vapor/sqlite-kit/releases/tag/4.4.2)**

This fixes an issue that arose with the release of SQLite 3.45.0, which includes support for a new "JSONB" internal representation. As a side effect, textual JSON data presented to SQLite as a BLOB is incorrectly treated as JSONB by the database, resulting in inexplicable errors when attempts are made to read the JSON back out again. Since we should always have been sending JSON to the database as TEXT in the first place, this is considered a general bugfix rather than purely a compatibility update.

Unblocks vapor/sqlite-nio#62.

### CODE REVIEW

**Review by 0xTim — APPROVED** [review]
Honestly surprised there wasn't more needed from turning on Sendable checking!

File: Sources/SQLiteKit/SQLiteConnection+SQLKit.swift [hunk 1]
```diff
@@ -82,7 +82,7 @@ internal struct _SQLiteDatabaseVersion: SQLDatabaseReportedVersion {
 }
 
 private struct _SQLiteSQLDatabase: SQLDatabase {
-    let database: SQLiteDatabase
+    let database: any SQLiteDatabase
```
Comment (MahdiBM) [inline Sources/SQLiteKit/SQLiteConnection+SQLKit.swift:85]: FYI they're dropping the ExistentialAny upcoming feature. I personally liked it. Not sure what exactly are their plans.

**Review by MahdiBM — APPROVED** [review]

File: Sources/SQLiteKit/SQLiteConnection+SQLKit.swift (hunk 1 above)
Comment (gwynne) [reply Sources/SQLiteKit/SQLiteConnection+SQLKit.swift:85]: Citation or it didn't happen 😜 What an absurd thing to do...

**Comment by gwynne** [comment]
> Honestly surprised there wasn't more needed from turning on Sendable checking!

@0xTim Well, it's not a fair test, at the moment - once SQLKit gets _its_ `Sendable` going (I've got most of it done in my local clone already), there'll be more warnings to solve in all the downstream dependencies (basically everything except async-kit and the three `-nio`s), even if they were "done" before.

**Pull request merged**